### PR TITLE
fix(routeToLocation): disable URI encoding

### PR DESCRIPTION
**Pull request by ya-woodcutter**
BREAKING: disable ALL URI encoding

### CODE REVIEW

File: src/parsers/routeToLocation.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ const createMatchRouteToPath = registry => ({ id, params = {}, query = {}, hash
     let pathname;
 
     try {
-        pathname = matcher(params, { pretty: true });
+        // path-to-regexp (1.6.0): encodeURI by default, disable it with decodeURI
+        // 'pretty' flag disable all encoding, besides '/', '?', '#'
+        pathname = decodeURIComponent(matcher(params));
```
Comment (evilj0e) [inline src/parsers/routeToLocation.js]: Why did you remove `{ pretty: true }`?

**Comment by ya-woodcutter** [comment]
Cause in `path-to-regexp` pretty ? encodeURIComponentPretty : encodeURIComponent
Then `decodeURIComponent(encodeURIComponent(content)) === content`

File: src/parsers/routeToLocation.js (hunk 1 above)
Comment (ya-woodcutter) [inline src/parsers/routeToLocation.js]: Cause in `path-to-regexp` pretty ? encodeURIComponentPretty : encodeURIComponent
Then `decodeURIComponent(encodeURIComponent(content)) === content`

Comment (evilj0e) [inline src/parsers/routeToLocation.js]: 👍

Comment (TimeRaider) [inline src/parsers/routeToLocation.js]: i'd love to see tests for this, but oh well

**Comment by ya-woodcutter** [comment]
@Blackheart340 @TimeRaider @evilj0e anyone can publish a new version?

**Comment by evilj0e** [comment]
@ya-woodcutter, done: 1.4.1
